### DISCUSSION TITLE
feature: Remember commit message when undoing commit

### DISCUSSION THREAD

**Discussion by Kamilcuk** [extrawurst/gitui#1872] [Q&A] [answered]
In git gui I can click "Amend last commit", then the last commit message will show up in the right down corner.

When I click "Undo Commit [U]" in gitui, then try to "Commit [c]", the commit message is empty, and I have to retype it.

**Describe the solution you'd like**
It would be great if the commit message would be remembered when undoing a commit. 

"Undo Commit [U]" in gitui is great, but it does not remember the commit message.

I am so much in love in "Amend Last Commit" in git gui. It is a very easy and flexible way for rebasing commits and splitting them into smaller parts. I would be happy to have this functionality in gitui.

**Describe alternatives you've considered**
git gui

**Additional context**
![image](https://github.com/extrawurst/gitui/assets/15679217/bb479a8b-8637-4ee8-9508-9bbbd77aeeb1)

Thanks for making gitui, it is just great.

(Additionally, subjectively, it would be great to show a one-line small window on right bottom below "Diff:" window in git ui to show the temporary commit message, just like git gui does. But I do not need it, this would only be "nice to have".)

**Comment by extrawurst** [comment, marked as answer]
You can use the commit msg history for that: default key bindings: ctrl-n

**Comment by Kamilcuk** [comment]
Like a charm! Thank you!
